ci/src/Registry/Scripts/BowerImport.purs

```
@@ -47,161 +45,279 @@ type PackageReleases = { address :: GitHub.Address, releases :: Set GitHub.Tag }
+-- | caching the file to avoid re-fetching each time the tool runs.
+fetchBowerfile :: RawPackageName -> GitHub.Address -> RawVersion -> ExceptT ImportError Aff Bower.PackageMeta
+fetchBowerfile name address tag = do
   let
-    targets = Foreign.fromFoldable $ Array.catMaybes
-      [ Just $ Tuple "lib"
-          { sources: [ "src/**/*.purs" ]
-          , dependencies: Foreign.fromFoldable deps
-          }
-      , if (List.null devDeps) then Nothing
-        else Just $ Tuple "test"
-          { sources: [ "src/**/*.purs", "test/**/*.purs" ]
-          , dependencies: Foreign.fromFoldable (deps <> devDeps)
-          }
-      ]
-
-  pure { name, license, repository, targets, version }
-
--- | Are all the dependencies PureScript packages or are there any external Bower/JS packages?
-selfContainedDependencies :: ReleasesIndex -> Bower.PackageMeta -> Maybe (Array String)
-selfContainedDependencies packageIndex (Bower.PackageMeta { dependencies, devDependencies }) =
+    url = "https://raw.githubusercontent.com/" <> address.owner <> "/" <> address.repo <> "/" <> un RawVersion tag <> "/bower.json"
+    bowerfileCache = "bowerfile__" <> un RawPackageName name <> "__" <> un RawVersion tag
+
+  withCache bowerfileCache Nothing do
+    lift (Http.get ResponseFormat.string url) >>= case _ of
+      -- TODO: We should retry requests if they fail, because likely GitHub
+      -- rate-limited us. Or at least we should collect the errors and print them
+      -- out because it's quite possible the Bowerfile actually does work and
+      -- we could have more granular errors to avoid excluding those packages.
+      Left err -> do
+        log $ "Unable to retrieve bowerfile. Bad request: " <> Http.printError err
+        throwError MissingBowerfile
+      Right { body, status }
+        | status /= StatusCode 200 -> do
+            log $ "Unable to retrieve bowerfile. Bad status: " <> body
+            throwError MissingBowerfile
```

I think we should check if we get a 404, and throw a `MissingBowerfile` only in that case - this is because every other code is temporary and we want to retry next time we run the script, rather than permanently skip the fetch